### ISSUE TITLE
PSDSEval.add_operating_point needs index column explicitly right now.

### ISSUE THREAD

**Issue by popcornell** [audioanalytic/psds_eval#3]
Hi, 

I don't know if this is intended but right now `PSDSEval.add_operating_point ` needs in input a DataFrame with an index column explicitly otherwise its behavior gets weird and some points are not added (only 4 out of 50). 
E.g 

```
      event_label  onset  offset                          filename
0   Vacuum_cleaner  0.000   9.856  Y00pbt6aJV8Y_350.000_360.000.wav
0   Vacuum_cleaner  0.000   9.856    Y00pK0GMmE9s_70.000_80.000.wav
0           Dishes  6.656   7.232    Y02sD1KJeoGA_50.000_60.000.wav
1           Dishes  9.728   9.856    Y02sD1KJeoGA_50.000_60.000.wav
2           Frying  0.000   9.856    Y02sD1KJeoGA_50.000_60.000.wav
..             ...    ...     ...                               ...
4           Speech  9.728   9.856    YOvK__OSdpJQ_74.000_84.000.wav
0           Speech  1.344   1.664  Yb8GxUkjLSUY_628.000_638.000.wav
1           Speech  1.984   2.560  Yb8GxUkjLSUY_628.000_638.000.wav
2           Speech  4.544   5.184  Yb8GxUkjLSUY_628.000_638.000.wav
0              Cat  2.176   2.560    Y86owBlJa8f0_24.000_34.000.wav
[2681 rows x 4 columns]
```
does not work, 

while

```
      Unnamed: 0     event_label  ...  offset                          filename
0              0  Vacuum_cleaner  ...   9.856  Y00pbt6aJV8Y_350.000_360.000.wav
1              0  Vacuum_cleaner  ...   9.856    Y00pK0GMmE9s_70.000_80.000.wav
2              0          Dishes  ...   7.232    Y02sD1KJeoGA_50.000_60.000.wav
3              1          Dishes  ...   9.856    Y02sD1KJeoGA_50.000_60.000.wav
4              2          Frying  ...   9.856    Y02sD1KJeoGA_50.000_60.000.wav
...          ...             ...  ...     ...                               ...
2805           2          Speech  ...   3.840  Yb8GxUkjLSUY_628.000_638.000.wav
2806           3          Speech  ...   5.248  Yb8GxUkjLSUY_628.000_638.000.wav
2807           4          Speech  ...   8.064  Yb8GxUkjLSUY_628.000_638.000.wav
2808           5          Speech  ...   9.344  Yb8GxUkjLSUY_628.000_638.000.wav
2809           0             Cat  ...   2.560    Y86owBlJa8f0_24.000_34.000.wav
[2810 rows x 5 columns]
```

works. 
The Unnamed column is added automatically by `pd.read_csv(tsv, sep="\t")`. 
You can reproduce this in the jupyter notebook example by setting 
  `  det = pd.read_csv(tsv, sep="\t", index_col=0)`. 
  My workaround has been now to do the following:
```python 
  det["index"] = range(1, len(det) + 1)
  det = det.set_index("index")
```
  I explicitly make the index column when a DataFrame does not have one. 
  I am using python 3.8.5 and version 0.3.0 of psds_eval.

**Comment by mralgos** [comment]
Thank you @popcornell for the raising the issue. 
May I ask you to provide an example of the detection table (i.e. the `tsv` file content) you are reading with `pd.read_csv`?

**Comment by popcornell** [comment]
I don't currently save predictions at all. 
But the format is this one: 

```
      event_label  onset  offset                          filename
0   Vacuum_cleaner  0.000   9.856  Y00pbt6aJV8Y_350.000_360.000.wav
0   Vacuum_cleaner  0.000   9.856    Y00pK0GMmE9s_70.000_80.000.wav
0           Dishes  6.656   7.232    Y02sD1KJeoGA_50.000_60.000.wav
1           Dishes  9.728   9.856    Y02sD1KJeoGA_50.000_60.000.wav
2           Frying  0.000   9.856    Y02sD1KJeoGA_50.000_60.000.wav
..             ...    ...     ...                               ...
4           Speech  9.728   9.856    YOvK__OSdpJQ_74.000_84.000.wav
0           Speech  1.344   1.664  Yb8GxUkjLSUY_628.000_638.000.wav
1           Speech  1.984   2.560  Yb8GxUkjLSUY_628.000_638.000.wav
2           Speech  4.544   5.184  Yb8GxUkjLSUY_628.000_638.000.wav
0              Cat  2.176   2.560    Y86owBlJa8f0_24.000_34.000.wav
[2681 rows x 4 columns]
```

while PSDSEval seems to expect something like: 

```
      Unnamed: 0     event_label  ...  offset                          filename
0              0  Vacuum_cleaner  ...   9.856  Y00pbt6aJV8Y_350.000_360.000.wav
1              0  Vacuum_cleaner  ...   9.856    Y00pK0GMmE9s_70.000_80.000.wav
2              0          Dishes  ...   7.232    Y02sD1KJeoGA_50.000_60.000.wav
3              1          Dishes  ...   9.856    Y02sD1KJeoGA_50.000_60.000.wav
4              2          Frying  ...   9.856    Y02sD1KJeoGA_50.000_60.000.wav
...          ...             ...  ...     ...                               ...
2805           2          Speech  ...   3.840  Yb8GxUkjLSUY_628.000_638.000.wav
2806           3          Speech  ...   5.248  Yb8GxUkjLSUY_628.000_638.000.wav
2807           4          Speech  ...   8.064  Yb8GxUkjLSUY_628.000_638.000.wav
2808           5          Speech  ...   9.344  Yb8GxUkjLSUY_628.000_638.000.wav
2809           0             Cat  ...   2.560    Y86owBlJa8f0_24.000_34.000.wav
[2810 rows x 5 columns]

```

to work properly.

**Comment by mralgos** [comment]
I see, so you don't save predictions in one file and then read them. You append each file predictions to a DataFrame and then pass it to the `add_operating_point` function. Am I right? 

If I'm right, have you tried using `df.reset_index(inplace=True)`? This won't require you to save and read with `index_col=0`?

I'm just trying to clarify the use case so that we can determine the appropriate fix/improvement.

**Comment by popcornell** [comment]
> I see, so you don't save predictions in one file and then read them. You append each file predictions to a DataFrame and then pass it to the add_operating_point function. Am I right?
exactly. 

`df.reset_index(inplace=True, drop=True)`

also works. 
It would be sufficient to raise an error in PSDSEval if possible if the indexes are off. Because right now you get a totally weird behaviour.

**Comment by mralgos** [comment]
@popcornell absolutely. Thanks for clarifying. 
We'll push a fix ASAP.

**Comment by calebmarchent** [comment]
@popcornell we've released a fix in release 0.4.0, please check it works for you.

**Comment by popcornell** [comment]
I ve checked and it seems fine now, thank you very much !
